### PR TITLE
Fixed crash on goToDef when a shorthand property refers to an undefined entity

### LINKED ISSUE #2384: Assert in dev mode with doing GotoDef on shorthand property
```typescript
let x = {
    foo
}
```

enable dev mode, put the cursor on `foo` and hit F12.
**Expected**: message box with text 'Cannot navigate to symbol under the caret'
**Actual**: Assert message box with the text

>---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
Error executing script:
Description:
getDefinitionAtPosition('C:\Users\vladima\AppData\Local\Temp\~vs4B6E.ts', 15)
Message:
StackTrace:
TypeError: Unable to get property 'getDeclarations' of undefined or null reference
   at getDefinitionAtPosition (:29601:17)
   at Anonymous function (:32196:17)
   at simpleForwardCall (:32043:9)
   at forwardJSONCall (:32057:13)
   at forwardJSONCall (:32088:13)
   at getDefinitionAtPosition (:32195:13)
   at Microsoft.CodeAnalysis.Editor.TypeScript.Errors.WERReport.CreateAndSubmitNonFatalReport(String description, String message, String stackTrace)
   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.JsonHelpers.CheckAndReportError(JToken json)
   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.JsonHelpers.TryGetResult[TToken](String encoded, TToken& result)
   at Microsoft.CodeAnalysis.Editor.TypeScript.ScriptServices.JsonHelpers.ParseNavigableItems(Project project, String encoded)
   at Microsoft.CodeAnalysis.Editor.TypeScript.S......
<truncated>
---------------------------
Abort   Retry   Ignore   
---------------------------

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #2384.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
:+1:

File: tests/cases/fourslash/goToDefinitionShorthandProperty03.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/// <reference path='fourslash.ts' />
+
+////let /*def*/x = {
+////    /*prop*/x
+////}
+
+goTo.marker("prop");
```
Comment (yuit) [inline tests/cases/fourslash/goToDefinitionShorthandProperty03.ts]: Should it go to the marker "def"? Shouldn't it just stay where it is

**Comment by yuit** [comment]
lgtm after you add a test cover for var.

**Comment by vladima** [comment]
:+1:

**Pull request merged**